app/HutchManager/Models/DataSource.cs

```diff
@@ -4,12 +4,15 @@ public class DataSource
 {
   public string Id { get; set; } = string.Empty;
   public DateTimeOffset LastCheckin { get; set; }
+  
+  public List<Data.Entities.Agent> Agents { get; set; }
 
   public DataSource(Data.Entities.DataSource entity)
   {
     Id = entity.Id;
     LastCheckin = entity.LastCheckin;
-   
+    Agents = entity.Agents;
```

Comment:
need to fix nullable warnings on this, but the source of them is the `null!` on the entity.

So either my suggestion there will fix it, or if EF isn't happy, you can change this to `entity.Agents!`